h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -1418,27 +1533,35 @@ private void fitLSM(Solver s){
 
     private void fitIRLSM(Solver s) {
       GLMWeightsFun glmw = new GLMWeightsFun(_parms);
-      double [] betaCnd = _state.beta();
+      double [] betaCnd = _checkPointFirstIter ? _model._betaCndTemp : _state.beta();
       LineSearchSolver ls = null;
-      boolean firstIter = true;
-      int iterCnt = 0;
+      int iterCnt = _checkPointFirstIter?_state._iter:0;
+      boolean firstIter = iterCnt == 0;
       try {
         while (true) {
           iterCnt++;
           long t1 = System.currentTimeMillis();
           ComputationState.GramXY gram = _state.computeGram(betaCnd,s);
           long t2 = System.currentTimeMillis();
-          if (!_state._lsNeeded && (Double.isNaN(gram.likelihood) || _state.objective(gram.beta, gram.likelihood) > _state.objective() + _parms._objective_epsilon)) {
+          if ((!_state._lsNeeded && (Double.isNaN(gram.likelihood) || _state.objective(gram.beta, gram.likelihood) > 
+                  _state.objective() + _parms._objective_epsilon)) && !_checkPointFirstIter) {
             _state._lsNeeded = true;
           } else {
-            if (!firstIter && !_state._lsNeeded && !progress(gram.beta, gram.likelihood)) {
-              Log.info("DONE after " + (iterCnt-1) + " iterations (1)");
+            if (!firstIter && !_state._lsNeeded && !progress(gram.beta, gram.likelihood) && !_checkPointFirstIter) {
+              Log.info("DONE after " + (iterCnt - 1) + " iterations (1)");
+              _model._betaCndTemp = s == Solver.COORDINATE_DESCENT ? COD_solve(gram, _state._alpha, _state.lambda())
```

Comment:
will this impact performance? this computation is done elswhere as well, but the result computed here is not used, could it be done so this is computed only once?

Reply:
This is done only once when the fitting is completed.  Yes, I think it will add one ADMM.solve.  I will make check the performance speed in the benchmarking.

Reply:
It will add one more ADMM_solve as it is done when it exits.  Will add performance test on my branch to measure the effect.